src/extension/networkInspector/networkInspectorServer.ts

```diff
@@ -0,0 +1,393 @@
+import { OutputChannelLogger } from "../log/OutputChannelLogger";
+import { Responder, Payload, ReactiveSocket } from "rsocket-types";
+import {
+    CertificateProvider,
+    SecureServerConfig,
+    CertificateExchangeMedium,
+} from "./certificateProvider";
+import { ClientOS } from "./clientUtils";
+import * as net from "net";
+import * as tls from "tls";
+import * as nls from "vscode-nls";
+import { InspectorViewType } from "./views/inspectorView";
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+const localize = nls.loadMessageBundle();
+
+/**
+ * @preserve
+ * Start region: the code is borrowed from https://github.com/facebook/flipper/blob/master/desktop/app/src/server.tsx
+ *
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+function transformCertificateExchangeMediumToType(
+    medium: number | undefined,
+): CertificateExchangeMedium {
+    if (medium == 1) {
+        return "FS_ACCESS";
+    } else if (medium == 2) {
+        return "WWW";
+    } else {
+        return "FS_ACCESS";
+    }
+}
+
+export const NETWORK_INSPECTOR_LOG_CHANNEL_NAME = "Network Inspector";
+
+export class NetworkInspectorServer {
+    public static readonly SecureServerPort = 8088;
+    public static readonly InsecureServerPort = 8089;
+
+    private connections: Map<string, ClientDevice>;
+    private secureServer: RSocketServer<any, any> | null;
+    private insecureServer: RSocketServer<any, any> | null;
+    private certificateProvider: CertificateProvider;
+    private initialisePromise: Promise<void> | null;
+    private logger: OutputChannelLogger;
+
+    constructor() {
+        this.connections = new Map<string, ClientDevice>();
+        this.logger = OutputChannelLogger.getChannel(NETWORK_INSPECTOR_LOG_CHANNEL_NAME);
+    }
+
+    public async start(adbHelper: AdbHelper): Promise<void> {
+        this.logger.info(localize("StartNetworkinspector", "Starting Network inspector"));
+        this.initialisePromise = new Promise(async (resolve, reject) => {
+            this.certificateProvider = new CertificateProvider(adbHelper);
+
+            try {
+                let options = await this.certificateProvider.loadSecureServerConfig();
+                this.secureServer = await this.startServer(
+                    NetworkInspectorServer.SecureServerPort,
+                    options,
+                );
+                this.insecureServer = await this.startServer(
+                    NetworkInspectorServer.InsecureServerPort,
+                );
+            } catch (err) {
+                return reject(err);
+            }
+
+            this.logger.info(localize("NetworkInspectorWorking", "Network inspector is working"));
+            resolve();
+        });
+        return await this.initialisePromise;
```

Comment:
Could you please remove await from the return statement? Because:

> Using return await just adds extra time before the overreaching promise is resolved without changing the semantics.

Sourсe: [tslint/rules/no-return-await/](https://palantir.github.io/tslint/rules/no-return-await/)